feature/text_based.py

```diff
@@ -0,0 +1,724 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: GNU GPLv3
+
+from typing import NoReturn, Union, List, Optional
+
+import random
+import pandas as pd
+import numpy as np
+from feature.base import _BaseSupervisedSelector
+from feature.utils import Num
+import matplotlib.pyplot as plt
+
+from mip import Model, xsum, minimize, maximize, BINARY, INTEGER, OptimizationStatus
+from sklearn.cluster import KMeans
+from scipy import sparse
+
+
+from textwiser import TextWiser, Embedding, Transformation
+
+
+
+
+class Data:
+    def __init__(self, cost: Union[List, np.ndarray], matrix: Union[List[List], np.ndarray]):
+        self._verify_input(cost, matrix)
+
+        # Cost Vector c[variable]
+        self.cost = cost
+
+        # Coefficient Matrix M[row][column]
+        self.matrix = matrix
+
+        # Range of variables
+        self.X = range(len(self.cost))
+
+        # Range of rows
+        self.rows = range(self.matrix.shape[0])
+
+        # Range of columns
+        self.columns = range(self.matrix.shape[1])
+
+    @staticmethod
+    def _verify_input(cost, matrix):
+        assert (cost is not None), "Data Error: cost cannot be none"
+        assert (matrix is not None), "Data Error: matrix cannot be none"
+        assert (matrix.ndim == 2), "Data Error: matrix should 2D"
+        assert (len(cost) == matrix.shape[1]), \
+            "Data Error: cost vector size should match the number of columns " + \
+            str(len(cost)) + " vs. " + str(matrix.shape[1])
+
+
+class ContentSelector:
+    """ Content Selector.
+
+    This class contains a content selection algorithm (max_cover) that maximizes the diversity among the selected
```

Comment:
I think this paragraphs need a bit of adjusting. This class does not contain a single max_cover algorithm, does it?

Reply:
yes, modified.